### ISSUE TITLE
Should leading ampersands (&) be allowed in intersection types?

### ISSUE THREAD

**Issue by Ailrun** [Microsoft/TypeScript#17916]
**TypeScript Version:**  2.4.2 / playground (2.3.?)

**Code**

```ts
let x: (& number & string);
```

**Expected behavior:**
Emits error, since [language spec](https://github.com/Microsoft/TypeScript/blob/master/doc/spec.md) said,

> *IntersectionType*: *IntersectionOrPrimaryType* & *PrimaryType*

in the [subsection 3.8.7](https://github.com/Microsoft/TypeScript/blob/master/doc/spec.md#3.8.7)

**Actual behavior:**
Doesn't emit error and uses it as `number & string`.

**Comment by yortus** [comment]
@Ailrun it is by design. See #12386. The spec might need updating though.

**Comment by Ailrun** [comment]
@yortus Why don't anyone try to fix the spec? Isn't specification of language the most important document?

**Comment by yortus** [comment]
The spec hasn't been updated since January 2016, so it's likely that a bunch of stuff is missing or outdated. I'm not sure why this is; my guess would be time pressure on the team, and the fact that they don't directly take PRs for the spec. Perhaps a team member can chime in.

**Comment by ikatyang** [comment]
See https://github.com/Microsoft/TypeScript/pull/17215#issue-243137359.

**Comment by Ailrun** [comment]
@yortus Alright... thank you for your answer!
@ikatyang Thank you for information!

**Comment by mhegazy** [comment]
Automatically closing this issue for housekeeping purposes. The issue labels indicate that it is unactionable at the moment or has already been addressed.
